stem/output.py

```diff
@@ -0,0 +1,219 @@
+            )
+            print(_msg)
+
+
+@dataclass
+class OutputParametersABC(ABC):
+    """
+    Abstract class for the definition of user output parameters (GiD, VTK, json).
+
+    Attributes:
+        -
+    """
+
+    pass
+
+    @abstractmethod
+    def validate(self):
+        """
+        Abstract method for validating user inputs
+        """
+        pass
+
+
+@dataclass
+class GiDOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        gidpost_flags (Dict[str, Any]):
+        file_label (str):
+        output_control_type (str):
+        output_interval (int):
+        body_output (bool):
+        node_output (bool):
+        skin_output (bool):
+        plane_output (List[str]):
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        point_data_configuration (List[str]):
+    """
+
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+
+    gid_post_mode: str = "GiD_PostBinary"
+    file_label: str = "step"
+    output_control_type: str = "step"
+    output_interval: int = 1
+    body_output: bool = True
+    node_output: bool = False
+    skin_output: bool = False
+    # optional
+    plane_output: List[str] = field(default_factory=lambda: [])
+    point_data_configuration: List[str] = field(default_factory=lambda: [])
+
+    def validate(self):
+        pass
+
+
+@dataclass
+class VtkOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        file_format (str): file format for VTK, either `binary` or `ascii` are allowed.
+        output_precision (int):
+        output_control_type (str): type of output control, either `step` or `time`.
+        output_interval (float):
+    """
+    # general inputs
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    # VTK specif inputs
+    file_format: str = "binary"
+    output_precision: int = 7
+    output_control_type: str = "step"
+    output_interval: float = 1.0
+
+    def validate(self):
+        detect_tensor_outputs(requested_outputs=self.gauss_point_results)
+
+
+@dataclass
+class JsonOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for JSON output
+
+    Attributes:
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        time_frequency (float):
+        sub_model_part_name (str):
+    """
+    # general inputs
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    # JSON specif inputs
+    time_frequency: float = 1.0
+    sub_model_part_name: str = ""
```

Comment:
To be checked

Reply:
It has been checked. "sub_model_part_name" is an optional parameter and it runs fine when not specified (using "").
When specified as part_name no changes are detected in the output (I guess because no subpart has been defined in the simple model).
I think we should keep it as input because the sub_model_part_name can be different than part_name.

Opinions?